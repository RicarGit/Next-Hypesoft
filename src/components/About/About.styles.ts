import styled from "styled-components"

export const AboutSection = styled.section`
  width: 100%;
  height: 795px;
  position: absolute;
  top: 1083px;
  display: flex;
`

export const Container = styled.div`
  width: 1320px;
  display: flex;
  margin: 0 auto;
  align-items: center;
`

export const ImageWrapper = styled.div`
  position: relative;
  display: flex;
  margin: 0;
  padding: 0;
  border-radius: 10px;
  box-shadow: 32px 32px white;
`

export const InfoContainer = styled.div`
  width: 672px;
  height: 484px;
  display: flex;
  flex-direction: column;
  margin-left: 88px;
  gap: 23px;
`

export const InfoTitle = styled.h2`
  font-size: 48px;
  font-weight: 500;
  line-height: 56px;
  color: ${({ theme }) => theme.colors.white};
`

export const InfoText = styled.p`
  font-size: 18px;
  line-height: 28px;
  color: ${({ theme }) => theme.colors.gray[500]};
`

export const ExperienceInfoContainer = styled.div`
  display: flex;
  width: 507px;
  height: 64px;
  gap: 50px;
`

export const ExperienceWrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 16px;
`

export const ExperienceNumber = styled.span`
  font-size: 56px;
  line-height: 64px;
  color: ${({ theme }) => theme.colors.secondary};
`

export const ExperienceText = styled.p`
  font-size: 18px;
  color: ${({ theme }) => theme.colors.white};
`