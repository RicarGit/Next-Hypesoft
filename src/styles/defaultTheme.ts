import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme