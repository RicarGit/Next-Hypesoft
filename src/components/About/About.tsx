import * as S from './About.styles'

import Image from 'next/image'
import { Assets } from '~/assets'

import { Button } from '~/shared/Button'

export const About = () => {
  return (
    <S.AboutSection id='about'>
      <S.Container>

        <S.ImageWrapper>
          <Image src={Assets.AboutImage} width={560} height={542} alt='about' />
        </S.ImageWrapper>

        <S.InfoContainer>
          <S.InfoTitle>
            Software Engineer and Infrastructure specialist based in Brazil.
          </S.InfoTitle>

          <S.InfoText>
            Morbi quam velit, euismod in imperdiet vitae, elementum et elit.
            Nunc finibus, felis sit amet sollicitudin sollicitudin, nisi magna feugiat enim, in maximus urna enim ac tortor.
            Nunc in volutpat ipsum, molestie commodo odio.
            Quisque auctor nisi mi. Aenean venenatis sapien et interdum interdum.
          </S.InfoText>

          <S.ExperienceInfoContainer>

            <S.ExperienceWrapper>
              <S.ExperienceNumber>17+</S.ExperienceNumber>
              <S.ExperienceText>Years of Experience</S.ExperienceText>
            </S.ExperienceWrapper>

            <S.ExperienceWrapper>
              <S.ExperienceNumber>325+</S.ExperienceNumber>
              <S.ExperienceText>Completed Projects</S.ExperienceText>
            </S.ExperienceWrapper>

          </S.ExperienceInfoContainer>

          <Button>Say Hi</Button>
        </S.InfoContainer>

      </S.Container>
    </S.AboutSection>
  )
}