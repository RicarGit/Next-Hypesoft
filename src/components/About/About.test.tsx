import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import About from '.'

describe('About', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <About />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})