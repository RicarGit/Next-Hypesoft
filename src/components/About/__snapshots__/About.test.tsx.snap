// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `
<section
  class="sc-bczRLJ ivBYWb"
  id="about"
>
  <div
    class="sc-gsnTZi eaBcyJ"
  >
    <div
      class="sc-dkzDqf eishpv"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27560%27%20height=%27542%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="about"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="sc-hKMtZM kKqypj"
    >
      <h2
        class="sc-eCYdqJ cnlhBJ"
      >
        Software Engineer and Infrastructure specialist based in Brazil.
      </h2>
      <p
        class="sc-jSMfEi iIiovf"
      >
        Morbi quam velit, euismod in imperdiet vitae, elementum et elit. Nunc finibus, felis sit amet sollicitudin sollicitudin, nisi magna feugiat enim, in maximus urna enim ac tortor. Nunc in volutpat ipsum, molestie commodo odio. Quisque auctor nisi mi. Aenean venenatis sapien et interdum interdum.
      </p>
      <div
        class="sc-gKXOVf XSpvr"
      >
        <div
          class="sc-iBkjds jxfwZc"
        >
          <span
            class="sc-ftvSup kFLCbd"
          >
            17+
          </span>
          <p
            class="sc-papXJ OXoXr"
          >
            Years of Experience
          </p>
        </div>
        <div
          class="sc-iBkjds jxfwZc"
        >
          <span
            class="sc-ftvSup kFLCbd"
          >
            325+
          </span>
          <p
            class="sc-papXJ OXoXr"
          >
            Completed Projects
          </p>
        </div>
      </div>
      <button
        class="sc-jqUVSM bovtQP"
      >
        <p
          class="sc-kDDrLX dRjXFK"
        >
          Say Hi
        </p>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="white arrow"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </button>
    </div>
  </div>
</section>
`;
